src/Lykke.AzureQueueIntegration/Publisher/AzureQueuePublisher.cs

```diff
@@ -32,18 +43,59 @@ public AzureQueuePublisher(string applicationName, AzureQueueSettings settings)
             _settings = settings;
             _settings.QueueName = _settings.QueueName.ToLower();
 
+            _cloudQueue = _settings.GetQueueAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+
             if (disableTelemetry)
                 DisableTelemetry();
         }
 
+        /// <summary>
+        /// Creates instance of the <see cref="AzureQueuePublisher{TModel}"/>
+        /// </summary>
+        /// <param name="logFactory">Log factory</param>
+        /// <param name="serializer">Serializer</param>
+        /// <param name="publisherName">Name of the publisher</param>
+        /// <param name="settings">Queue settings</param>
+        /// <param name="bufferLifetime">Time interval, which indicates how often buffered messages will be published to the queue</param>
+        /// <param name="disableTelemetry">Disables Application Insight telemetry</param>
+        public AzureQueuePublisher(
+            [NotNull] ILogFactory logFactory,
+            [NotNull] IAzureQueueSerializer<TModel> serializer,
+            [NotNull] string publisherName,
+            [NotNull] AzureQueueSettings settings,
+            TimeSpan? bufferLifetime = null,
+            bool disableTelemetry = true)
+            
+            : base(bufferLifetime ?? TimeSpan.FromMilliseconds(100), logFactory, publisherName)
+        {
+            _serializer = serializer ?? throw new ArgumentNullException(nameof(serializer));
+            _settings = settings ?? throw new ArgumentNullException(nameof(settings));
+            _log = logFactory?.CreateLog(this, publisherName) ?? throw new ArgumentNullException(nameof(logFactory));
```

Comment:
https://github.com/LykkeCity/CommonDotNetLibraries/pull/58/files#diff-785f2378c107258fd49a1c495bb10ea7R65
The base class throws an exception early.

Reply:
I would use NullLog as a default. Without ?.logSomething() in code

Reply:
I've removed ```throw```, but ```NullLogFactory``` usage is not necessary here, since ```logFactory``` is ```[NotNull]``` and checked for null in the base class.